src/ultk/language/grammar/grammar.py

```diff
@@ -19,6 +20,7 @@
 from ultk.util.frozendict import FrozenDict
 
 T = TypeVar("T")
+Dataset = Iterable[tuple[Referent, T]]
```

Comment:
I think this one and `Iterable` import are no longer needed in this particular file